pwndbg/arguments.py

```diff
@@ -90,12 +84,9 @@ def get(instruction: PwndbgInstruction) -> List[Tuple[pwndbg.lib.functions.Argum
 
     func = pwndbg.lib.functions.functions.get(name, None)
 
-    # Try to extract the data from GDB.
-    # Note that this is currently broken, pending acceptance of
-    # my patch: https://sourceware.org/ml/gdb-patches/2015-06/msg00268.html
-    if sym and sym[0]:
+    if sym:
         try:
-            n_args_default = len(sym[0].type.fields())
+            n_args_default = len(sym.type.fields())
```

Comment:
`len(sym.type.fields())` returns 0 if functions is found, but no debug symbols...
Should how should we fallback? Maybe we should create another method for that?
`sym.type.func_arguments() -> List[Type] | None` and detect no symbols and raise exception? or return None

Only for gdb:
```
In [26]: sym.type
Out[26]: <gdb.Type code=TYPE_CODE_FUNC name=<text variable, no debug info>>

In [28]: sym.type.name
Out[28]: '<text variable, no debug info>'
```

Reply:
in previous version `gdb.lookup_symbol` returns None when function don't have debug info.
And in new version with `aglib.symbol.lookup_frame_symbol` we returns `*function` address, but some data is missing like function argument (due to missing debug info - when debugging binary without debug info)

Reply:
https://github.com/bminor/binutils-gdb/blob/c2dbc2929e87557f8bc030f6f010d67b19f99f12/gdb/gdbtypes.c#L6052-L6072

Reply:
Fixed